### PR TITLE
Export import unit tests

### LINKED ISSUE #2619: Tests are lost when exported
**Describe the bug**
Exporting a workspace appears to lose the tests that have been created.

**To Reproduce**
1. Create a new OAS document
2. Create test suites and some tests
3. Go to preferences >> Data >> Export >> Current workspace >> All requests
4. Delete the original document
5. Go to preferences >> Data >> Import >> From File >> Import the document that was just exported
6. Go back to tests and there will be no tests imported

**Expected behavior**
The imported file should bring over the tests that I have created.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. macOS 10.15.5 ]
 - Installation Method: [Download]
 - App Version [2020.3.3]

**Additional context**

### PR DESCRIPTION
**Pull request by sonicyeti**
This PR covers the importing & exporting of unit test suites, tests and results. Video demo can be seen here:

https://share.getcloudapp.com/4guJQJjQ

Closes: #2619 
FIXES INS-75

### CODE REVIEW

File: packages/insomnia-app/app/common/__tests__/import.test.js [hunk 1]
```diff
@@ -278,6 +278,18 @@ describe('export', () => {
...
+      parentId: w._id,
+    });
+    const ut1 = await models.unitTest.create({
+      name: 'Unit Test One',
+      parentId: uts1._id,
+    });
+    const utr1 = await models.unitTestResult.create({
+      name: 'Unit Test 1 Result 1',
+      parentId: ut1._id,
+    });
```
Comment (develohpanda) [inline packages/insomnia-app/app/common/__tests__/import.test.js]: Unit test results in designer are similar to unit test results in our code - they're usually "in-memory" and not persisted when you clone a repo, or import, etc.

The same way that responses aren't exported (afaik), test results shouldn't be either

File: packages/insomnia-app/app/ui/components/wrapper-unit-test.js [hunk 2]
```diff
@@ -320,30 +320,47 @@ class WrapperUnitTest extends React.PureComponent<Props, State> {
       );
     }
 
-    const { stats, tests } = activeUnitTestResult.results;
+    // On imported spec
+    /*
+    if (!activeUnitTestResult.results) {
+      this._handleRunTests();
+    }
+    */
```
Comment (develohpanda) [inline packages/insomnia-app/app/ui/components/wrapper-unit-test.js]: I like the thought process here but in the current state there are some gotchas.

When importing, the environment could be set to whatever is in your spec, but lets say the environment was automatically set to _production_. We probably don't (ever) want to automatically run tests.

File: packages/insomnia-app/app/ui/components/wrapper-unit-test.js [hunk 3]
```diff
@@ -320,30 +320,47 @@ class WrapperUnitTest extends React.PureComponent<Props, State> {
...
-              )}
-            </div>
-            <ListGroup>
-              {tests.map((t, i) => (
-                <UnitTestResultItem key={i} item={t} />
-              ))}
-            </ListGroup>
+        <div>
+          <div className="unit-tests__top-header">
+            <h2 className="success">Awaiting Test Execution</h2>
```
Comment (develohpanda) [inline packages/insomnia-app/app/ui/components/wrapper-unit-test.js:354]: 👍 I like the placeholder, getting close to a nicer empty state

**Review by develohpanda — CHANGES_REQUESTED** [review]
Nicely done 😄 I don't think we should be exporting unit test results, though, so a small change to make please!

File: packages/insomnia-app/app/common/import.js [hunk 4]
```diff
@@ -339,13 +343,20 @@ export async function exportWorkspacesData(
...
+
+  return exportRequestsData(requests, includePrivateDocs, format, unitTestSuites, unitTests);
 }
 
 export async function exportRequestsData(
   requests: Array<BaseModel>,
   includePrivateDocs: boolean,
   format: 'json' | 'yaml',
+  unitTestSuites: Array<BaseModel>,
+  unitTests: Array<BaseModel>,
```
Comment (develohpanda) [inline packages/insomnia-app/app/common/import.js]: Doesn't look like these arguments are used - VS Code or ES Lint should probably detect it. Do they still need to exist? 😄

**Review by develohpanda — COMMENTED** [review]
Small comment, almost there!

File: packages/insomnia-app/app/common/import.js (hunk 4 above)
Comment (sonicyeti) [reply packages/insomnia-app/app/common/import.js]: Thanks for the catch!

File: packages/insomnia-app/app/common/import.js [hunk 5]
```diff
@@ -339,7 +343,12 @@ export async function exportWorkspacesData(
 ): Promise<string> {
   const docs: Array<BaseModel> = await getDocWithDescendants(parentDoc, includePrivateDocs);
   const requests: Array<BaseModel> = docs.filter(doc => doc.type === models.request.type);
-  return exportRequestsData(requests, includePrivateDocs, format);
+  const unitTestSuites: Array<BaseModel> = docs.filter(
+    doc => doc.type === models.unitTestSuite.type,
+  );
+  const unitTests: Array<BaseModel> = docs.filter(doc => doc.type === models.unitTest.type);
+
+  return exportRequestsData(requests, includePrivateDocs, format, unitTestSuites, unitTests);
```
Comment (develohpanda) [inline packages/insomnia-app/app/common/import.js]:
```suggestion
  return exportRequestsData(requests, includePrivateDocs, format);
```

**Review by develohpanda — APPROVED** [review]

**Pull request merged**